dht/i_partitioner.cc

```diff
@@ -295,43 +295,43 @@ bool ring_position::less_compare(const schema& s, const ring_position& other) co
     return tri_compare(s, other) < 0;
 }
 
-int ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
+std::strong_ordering ring_position_tri_compare(const schema& s, ring_position_view lh, ring_position_view rh) {
     auto token_cmp = tri_compare(*lh._token, *rh._token);
     if (token_cmp != 0) {
-        return token_cmp < 0 ? -1 : 1;
+        return token_cmp;
     }
     if (lh._key && rh._key) {
         auto c = lh._key->legacy_tri_compare(s, *rh._key);
         if (c) {
-            return c;
+            return c <=> 0;
         }
-        return lh._weight - rh._weight;
+        return (lh._weight - rh._weight) <=> 0;
     }
     if (!lh._key && !rh._key) {
-        return lh._weight - rh._weight;
+        return lh._weight - rh._weight <=> 0;
     } else if (!lh._key) {
-        return lh._weight > 0 ? 1 : -1;
+        return lh._weight > 0 ? std::strong_ordering::greater : std::strong_ordering::less;
     } else {
-        return rh._weight > 0 ? -1 : 1;
+        return rh._weight > 0 ? std::strong_ordering::less : std::strong_ordering::greater;
     }
 }
 
-int ring_position_comparator_for_sstables::operator()(ring_position_view lh, sstables::decorated_key_view rh) const {
+std::strong_ordering ring_position_comparator_for_sstables::operator()(ring_position_view lh, sstables::decorated_key_view rh) const {
     auto token_cmp = tri_compare(*lh._token, rh.token());
     if (token_cmp != 0) {
-        return token_cmp < 0 ? -1 : 1;
+        return token_cmp;
     }
     if (lh._key) {
         auto rel = rh.key().tri_compare(s, *lh._key);
         if (rel) {
-            return -rel;
+            return -rel < 0 ? std::strong_ordering::less : std::strong_ordering::greater;
```

Comment:
I was curious if there's a less ugly way to do this, and I think there is:
return 0 <=> rel;
Note the zero is first, so the directions are reversed.
EDIT: I see you used exactly this trick a few lines down.

Reply:
Yes, I'll change this too.

Reply:
Changed.